src/ray/gcs/store_client/redis_store_client.cc

```diff
@@ -244,16 +218,115 @@ Status RedisStoreClient::AsyncMultiGet(
   RAY_CHECK(callback);
   if (keys.empty()) {
     callback({});
+    return Status::OK();
   }
   std::vector<std::string> true_keys;
   for (auto &key : keys) {
     true_keys.push_back(GenRedisKey(external_storage_namespace_, table_name, key));
   }
-  RAY_CHECK_OK(MGetValues(
-      redis_client_, external_storage_namespace_, table_name, true_keys, callback));
+  MGetValues(table_name, true_keys, callback);
   return Status::OK();
 }
 
+size_t RedisStoreClient::PushToSendingQueue(const std::vector<std::string> &keys,
+                                            std::function<void()> send_request) {
+  size_t queue_added = 0;
+  for (const auto &key : keys) {
+    auto [op_iter, added] =
+        pending_redis_request_by_key_.emplace(key, std::queue<std::function<void()>>());
+    if (added) {
+      queue_added++;
+    }
+    if (added) {
+      // There is no in-flight requests for this key which means
+      // in the callback the queue won't be poped. So just push
```

Comment:
nit:
As an optimization, if there is no in-flight request in this queue, we don't need to store the actual send_request in the queue but just need a placeholder (to indicate there are pending requests). This is because either 
1. the send_request will be fired immediately (if all the depending queues are empty).
2. otherwise the send_request in the last queue with pending in-flight requests will be called.
In either case, the send_request will not be called in this queue.